### PR TITLE
Fixed GCE playbooks so that they're more like the AWS playbooks.

### PR DESCRIPTION
**Pull request by twiest**
Namely the GCE playbooks now:
- Create infra nodes
- Correctly label nodes in OpenShift
- Setup masters as nodes as well (needed for sdn)
- Removed set_infra_launch_facts_tasks.yml as it's not used anymore.

### CODE REVIEW

**Comment by twiest** [comment]
@menren @sdodson Hey Guys, mind looking at this please?

It should only affect GCE, and I tested it using the online deployment type against GCE and everything is working well for me.

@menren This is the code that enables infra nodes that I talked about in PR #658. Please make sure this doesn't break you. This is meant to make the GCE code work more like the AWS code.

File: playbooks/gce/openshift-cluster/tasks/launch_instances.yml [hunk 1]
```diff
@@ -20,22 +20,57 @@
       - host-type-{{ type }}
       - sub-host-type-{{ g_sub_host_type }}
       - env-host-type-{{ cluster }}-openshift-{{ type }}
-  when: instances |length > 0 
+  when: instances |length > 0
   register: gce
 
+- debug: var=gce
```
Comment (sdodson) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: Remove the debug?

File: playbooks/gce/openshift-cluster/tasks/launch_instances.yml [hunk 2]
```diff
@@ -20,22 +20,57 @@
...
+                 \n/var/log/spooler \n"
+          options:
+            - daily
+            - rotate 7
+            - compress
+            - sharedscripts
+            - missingok
+          scripts:
+            postrotate: "/bin/kill -HUP `cat /var/run/syslogd.pid 2> /dev/null` 2> /dev/null || true"
+
```
Comment (sdodson) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: Followup issue to move this out of GCE/AWS specific playbooks and into more a common playbook?

File: playbooks/gce/openshift-cluster/tasks/launch_instances.yml [hunk 3]
```diff
@@ -20,22 +20,57 @@
...
-  when: instances |length > 0 
+  when: instances |length > 0
   register: gce
 
+- debug: var=gce
+
+- set_fact:
+    node_label:
+      # There doesn't seem to be a way to get the region directly, so parse it out of the zone.
+      region: "{{ gce.zone | regex_replace('^(.*)-.*$', '\\\\1') }}"
```
Comment (menren) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: Fail at this step while deploying with 0 infra node.
In fact we skip the launch of instead in the code above, however this code is executed for the infra.
We should check again that `instances|length >0`

File: playbooks/gce/openshift-cluster/tasks/launch_instances.yml [hunk 4]
```diff
@@ -20,22 +20,57 @@
...
+      type: "{{ g_sub_host_type }}"
+  when: type == "node"
+
+- set_fact:
+    node_label:
+      # There doesn't seem to be a way to get the region directly, so parse it out of the zone.
```
Comment (menren) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: Fail at this step while deploying with 0 infra node.
In fact we skip the launch of instead in the code above, however this code is executed for the infra.
We should check again that `instances|length >0`

**Comment by menren** [comment]
As I mentionned, we need to do some check in the case we deploy without infra-node.

It seems good to me with infra-node : 
`
PLAY RECAP ******************************************************************** 
github-master-8fc35        : ok=173  changed=56   unreachable=0    failed=0   
github-node-compute-91dfe  : ok=61   changed=24   unreachable=0    failed=0   
github-node-compute-bb1fb  : ok=61   changed=24   unreachable=0    failed=0   
github-node-infra-25513    : ok=61   changed=24   unreachable=0    failed=0   
localhost                  : ok=49   changed=6    unreachable=0    failed=0  
`

File: playbooks/gce/openshift-cluster/tasks/launch_instances.yml (hunk 2 above)
Comment (detiber) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: :+1: on moving to a more common playbook in the future

Comment (twiest) [inline playbooks/gce/openshift-cluster/tasks/launch_instances.yml]: @sdodson @detiber   Agreed. I hate that this is even in here, but it was in the AWS one, so I copied it here.

I've created this issue to track it (as suggested by @sdodson):
https://github.com/openshift/openshift-ansible/issues/690

**Comment by twiest** [comment]
@menren @sdodson Please review again, I've addressed all comments.

@menren please also make sure it doesn't break you.

I also tested with and without infra nodes.

With infra nodes:
```
PLAY RECAP ******************************************************************** 
localhost                  : ok=43   changed=6    unreachable=0    failed=0   
twiesttest2-master-ff011   : ok=166  changed=56   unreachable=0    failed=0   
twiesttest2-node-compute-34b42 : ok=61   changed=24   unreachable=0    failed=0   
twiesttest2-node-infra-9cbf6 : ok=61   changed=24   unreachable=0    failed=0   
```

Without infra nodes:
```
PLAY RECAP ******************************************************************** 
localhost                  : ok=37   changed=4    unreachable=0    failed=0   
twiesttest3-master-ac34a   : ok=166  changed=56   unreachable=0    failed=0   
twiesttest3-node-compute-ba3c1 : ok=61   changed=24   unreachable=0    failed=0   

```

**Comment by menren** [comment]
Good for me !

**Comment by sdodson** [comment]
LGTM

**Pull request merged**